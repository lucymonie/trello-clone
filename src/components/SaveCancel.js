import React from 'react';

const SaveCancel = ({ onClickCancel, onClickSave }) => (
<div>
  <span className="save-button btn-small green" onClick={ onClickSave }>Save</span>
  <span className="cancel-button cursor-pointer" onClick={ onClickCancel }><i className="close-icon grey-text material-icons">close</i></span>
</div>
)

export default SaveCancel;
