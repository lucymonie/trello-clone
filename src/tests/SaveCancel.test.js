import React from 'react';
import { shallow } from 'enzyme';
import SaveCancel from '../components/SaveCancel';


describe("SaveCancel", () => {
  it("renders consistently", () => {
    const component = shallow(<SaveCancel />);
    expect(component).toMatchSnapshot();
  });

  describe("when clicking the Save button", () => {
    it("triggers the onClick", () => {
      const mockEvent = jest.fn();
      const component = shallow(<SaveCancel onClickSave={mockEvent} />);
      component.find('button').simulate("click");
      expect(mockEvent.mock.calls.length).toEqual(1);
    });
  });

  describe("when clicking the Cancel button", () => {
    it("triggers the onClick", () => {
      const mockEvent = jest.fn();
      const component = shallow(<SaveCancel onClickCancel={mockEvent} />);
      component.find('span').simulate("click");
      component.find('span').simulate("click");
      expect(mockEvent.mock.calls.length).toEqual(2);
    });
  });
});
