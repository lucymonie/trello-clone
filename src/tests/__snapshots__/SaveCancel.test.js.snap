// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveCancel renders consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaveCancel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="save-button btn-small green"
          onClick={undefined}
>
          Save
</button>,
        <span
          className="cancel-button cursor-pointer"
          onClick={undefined}
>
          <i
                    className="close-icon grey-text material-icons"
          >
                    close
          </i>
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save",
          "className": "save-button btn-small green",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Save",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="close-icon grey-text material-icons"
>
            close
</i>,
          "className": "cancel-button cursor-pointer",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "close-icon grey-text material-icons",
          },
          "ref": null,
          "rendered": "close",
          "type": "i",
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="save-button btn-small green"
            onClick={undefined}
>
            Save
</button>,
          <span
            className="cancel-button cursor-pointer"
            onClick={undefined}
>
            <i
                        className="close-icon grey-text material-icons"
            >
                        close
            </i>
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "className": "save-button btn-small green",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="close-icon grey-text material-icons"
>
              close
</i>,
            "className": "cancel-button cursor-pointer",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "close",
              "className": "close-icon grey-text material-icons",
            },
            "ref": null,
            "rendered": "close",
            "type": "i",
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
